unity/unity-embed-host/CoreCLRHost.cs

```diff
@@ -95,6 +95,34 @@ public static IntPtr gchandle_get_target_v2([NativeWrapperType("uintptr_t")] Int
         return Unsafe.As<object, IntPtr>(ref obj);
     }
 
+    [return: NativeCallbackType("MonoObject*")]
+    public static IntPtr object_isinst([NativeCallbackType("MonoObject*")] IntPtr obj, [NativeCallbackType("MonoClass*")] IntPtr klass)
+    {
+        var instance = Unsafe.As<IntPtr, object>(ref obj);
+        var type = Unsafe.As<IntPtr, Type>(ref klass);
+        Type current = instance.GetType();
+
+        while (current != null)
+        {
+            if (current == type)
+                return obj;
+
+            // TODO : Unclear what the value of this is.  The if above is always hit first preventing rank differences from being taken into consideration
```

Comment:
I haven't figured out what needs to happen here yet.  Based on reading the native implmentation https://github.com/Unity-Technologies/runtime/blob/20cf508e82ba7cce3ccb141acdaef2a3d0c7d4ad/src/coreclr/vm/mono/mono_coreclr.cpp#L969

I thought that test cases such as `[TestCase(typeof(Mammal), 0, typeof(Mammal), 1, false)]` should fail due to the rank being different.  However,

1) In the managed implementation the `if (current == type)` above is hit during this case and we say they are equal
2) I don't see any tests in `main.cpp` that test the array behavior of the current native implementation.